templates/next/components/common/Pagination.tsx

```diff
@@ -0,0 +1,45 @@
+import Link from "next/link";
+import { PagedCollection } from "../../types/Collection";
+
+interface Props {
+  collection: PagedCollection<unknown>;
+}
+
+const Pagination = ({ collection }: Props) => {
+  const view = collection && collection['{{{hydraPrefix}}}view'];
+  if (!view) return;
+
+  const {
+    '{{{hydraPrefix}}}first': first,
+    '{{{hydraPrefix}}}previous': previous,
+    '{{{hydraPrefix}}}next': next,
+    '{{{hydraPrefix}}}last': last
+  } = view;
+
+  return (
+    <nav aria-label="Page navigation">
+      <Link href={first ? first : '#'}>
```

Comment:
Wouldn't it be more correct to not generate the HTML at all if the page does't exist?

Reply:
That's already done [here](https://github.com/api-platform/client-generator/pull/292/files/14371f99802246fcde9c645547e160f7011a5683#diff-94912163717e897f12704ce6e8a866c0d2c78a0fa9cf747bd6361b20f1311f5aR10):
```
if (!view) return;
```